dcore/forcing.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta, abstractmethod
+
+class Forcing(object):
+    """
+    Base class for forcing terms for dcore.
+
+    :arg state: x :class:`.State` object.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, state):
+        self.state = state
+    
+    @abstractmethod
+    def apply(self,scale,):
+        """
+        Function takes x as input, computes F(x) and returns 
+        x_out = x + scale*F(x) 
+        as output.
+
+        :arg scale: parameter to scale the output by.
+        :arg x: :class:`.Function` object, the input Function.
+        :arg x_out: :class:`.Function` object, the output Function.
+        """
+        pass
```

Comment:
I should probably call it forcing-and-sourcing

On 21 January 2016 at 16:49, Colin Cotter <colin.cotter@imperial.ac.uk>
wrote:

> No, because x is the u+rho+theta, and we only solve for forcing for u
> here. Other forcing might occur in theta or rho.
>
> On 21 January 2016 at 16:47, Lawrence Mitchell <notifications@github.com>
> wrote:
>
>> In dcore/forcing.py
>> <https://github.com/firedrakeproject/dcore/pull/9#discussion_r50427135>:
>>
>> > +
>> > +    def __init__(self, state):
>> > +        self.state = state
>> > +
>> > +    @abstractmethod
>> > +    def apply(self,scale,):
>> > +        """
>> > +        Function takes x as input, computes F(x) and returns
>> > +        x_out = x + scale*F(x)
>> > +        as output.
>> > +
>> > +        :arg scale: parameter to scale the output by.
>> > +        :arg x: :class:`.Function` object, the input Function.
>> > +        :arg x_out: :class:`.Function` object, the output Function.
>> > +        """
>> > +        pass
>>
>> If the forcing is always, compute f(x), and add it with some scaling, can
>> the implementation of apply not live here?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/firedrakeproject/dcore/pull/9/files#r50427135>.
>>
>
>
>
> --
> http://www.imperial.ac.uk/people/colin.cotter
>
> www.cambridge.org/9781107663916
>
>
>


-- 
http://www.imperial.ac.uk/people/colin.cotter

www.cambridge.org/9781107663916